### PR TITLE
Add driver for GR716 pinmux

### PR DESCRIPTION
**Pull request by ghost**
This adds a pinmux driver allowing to configure the I/O function
select registers found on the GR716 MCU.

### CODE REVIEW

File: drivers/pinmux/pinmux_gr716.c [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+static int input(const struct device *dev, uint32_t pin, uint8_t func)
+{
+	return -ENOTSUP;
+}
+
+static int init(const struct device *dev)
+{
+	return 0;
```
Comment (galak) [inline drivers/pinmux/pinmux_gr716.c:157]: Why have this if its empty?

Comment (mbolivar-nordic) [reply drivers/pinmux/pinmux_gr716.c:157]: z_sys_init_run_level() doesn't null-check the init function, so this looks required to me.

Comment (galak) [reply drivers/pinmux/pinmux_gr716.c:157]: gotcha

File: include/dt-bindings/pinctrl/gr716-pinctrl.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_GR716_PINCTRL_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_GR716_PINCTRL_H_
+
+#define GR716_IO_MODE_MAX               0x000d
+#define GR716_IO_MODE_MASK              0x000f
+
+#define GR716_IO_PULL_MASK              0x0300
```
Comment (galak) [inline include/dt-bindings/pinctrl/gr716-pinctrl.h:13]: Please use dts/bindings/pinctrl/pincfg-node.yaml for common flags like pull up/pull down

**Comment by galak** [comment]
Marking this DNM in hopes that we get the new pinctrl api sorted out.

**Pull request closed without merging**